ls1intum/Artemis
# `Assessment`: Improve user experience during repository export

### Checklist
#### General



#### Client


### Motivation and Context
This PR closes #7158. The "Anonymize Repository" Flag has been checked once the export is submitted, regardless of what the user checked the box before. This is bad UX. This PR makes the UI with the used flags consistent and simplifies the conditions by allowing instructors to change and see the "anonymize" flag, whereas tutors don't see it and always have an anonymized repository

### Description
This PR removes the force-changed export options from the exportRepos() method to make the UI consistent with the actual behavior of the export. Also the condition is simplified (and slightly changed), so that Instructors are allowed to change the anonymize flag and tutors are not.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis as a Student
2. Submit a solution for the Programming exercise
3. Log in to Artemis as an instructor
4. Change The "Due Date" to now
5. Assess the submition
6. Click "Export Repository"
7. Verify that you see the checkbox for "Anonymize Repository"
8. Verify that the checkbox stays checked/unchecked depending on what you chose once submitted and that the downloaded Repository is anonymous/not anonymous depending on your choice.

9. Log out and log in as Tutor and verify that you don't see the "Anonymize Repository" Flag
10. Export the repository and see that it is anonymized.

### Review Progress







#### Code Review
#### Manual Tests


### Test Coverage







### Screenshots
Instructor:
![image](https://github.com/ls1intum/Artemis/assets/78978542/725e1020-89b3-4c70-84cc-3480fcc461f6)

Tutor:
![image](https://github.com/ls1intum/Artemis/assets/78978542/5bd4911e-a0d1-48c2-ae8b-a9a000118be8)
